### PR TITLE
[ONNX] Enable OneHot operation

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    {
+                        axis += indices_shape.size() + 1;
+                    }
+
+                    ASSERT_VALID_ARGUMENT(node, (axis >= 0) && (axis <= indices_shape.size()))
+                        << "provided 'axis' attribute is not valid.";
+
+                    auto cdepth = std::dynamic_pointer_cast<ngraph::op::Constant>(depth);
+
+                    ASSERT_VALID_ARGUMENT(node, cdepth) << "Only static depth is supported";
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: What does "static" mean in this context. Please use a better error message. Perhaps 
> "Only constant values for `depth` input are supported for the OneHot operator."

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+                    if (axis < 0)
+                    {
+                        axis += indices_shape.size() + 1;
+                    }
+
+                    ASSERT_VALID_ARGUMENT(node, (axis >= 0) && (axis <= indices_shape.size()))
+                        << "provided 'axis' attribute is not valid.";
+
+                    auto cdepth = std::dynamic_pointer_cast<ngraph::op::Constant>(depth);
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: Please change to `constant_depth` or `depth_constant`

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+                    ASSERT_VALID_ARGUMENT(node, cdepth) << "Only static depth is supported";
+
+                    std::int64_t depth_value = cdepth->get_vector<std::int64_t>()[0];
+                    auto output_shape = indices_shape;
+                    output_shape.insert(std::next(std::begin(output_shape), axis), depth_value);
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: Please add a comment explaning this line. What is the desired `output_shape`

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+                    std::shared_ptr<ngraph::Node> one_hot =
+                        std::make_shared<ngraph::op::OneHot>(indices, output_shape, axis);
+                    on_value = numpy_style_broadcast_for_binary_operation(one_hot, on_value)[1];
+                    off_value = numpy_style_broadcast_for_binary_operation(one_hot, off_value)[1];
+                    one_hot = one_hot * (on_value - off_value) + off_value;
+                    return {std::make_shared<ngraph::op::Convert>(
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: According to the doc, the type of the output tensor must be the same as the type of the values input tensor.

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                        element::i64);
+                    auto axis = node.get_attribute_value<int64_t>("axis", -1);
+
+                    if (axis < 0)
+                    {
+                        axis += indices_shape.size() + 1;
+                    }
+
+                    ASSERT_VALID_ARGUMENT(node, (axis >= 0) && (axis <= indices_shape.size()))
+                        << "provided 'axis' attribute is not valid.";
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: Please add the op type `OneHot` and the value of `axis` to the error message.

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                NodeVector onehot(const Node& node)
+                {
+                    NodeVector inputs{node.get_ng_inputs()};
+                    auto indices =
+                        std::make_shared<ngraph::op::Convert>(inputs.at(0), element::i64);
+                    auto indices_shape = indices->get_shape();
+                    auto depth = inputs.at(1);
+                    auto values = inputs.at(2);
+                    std::shared_ptr<ngraph::Node> off_value = std::make_shared<ngraph::op::Convert>(
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: On/off values may also be floating point numbers.

File: src/ngraph/frontend/onnx_import/op/onehot.cpp (hunk 5 above)
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/onehot.cpp]: Op type is already in error message, axis value will be added.
```
C++ exception with description "Assertion '(axis >= 0) && (axis <= indices_shape.size())' failed at /home/tsocha/dev/ngraph/src/ngraph/frontend/onnx_import/op/onehot.cpp:58:
<Node(OneHot): y> provided 'axis' attribute is not valid." thrown in the test body.
```

File: src/ngraph/frontend/onnx_import/op/onehot.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                    auto indices =
+                        std::make_shared<ngraph::op::Convert>(inputs.at(0), element::i64);
+                    auto indices_shape = indices->get_shape();
+                    auto depth = inputs.at(1);
+                    auto values = inputs.at(2);
+                    std::shared_ptr<ngraph::Node> off_value =
+                        std::make_shared<ngraph::op::Slice>(values, Coordinate{0}, Coordinate{1});
+                    std::shared_ptr<ngraph::Node> on_value =
+                        std::make_shared<ngraph::op::Slice>(values, Coordinate{1}, Coordinate{2});
+                    auto axis = node.get_attribute_value<int64_t>("axis", -1);
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/onehot.cpp]: Forgot `std::` prefix. Add also `#include <cstddef>`

**Pull request merged**